File: Notificare.Geo.Core/Models/NotificareVisit.cs
namespace NotificareSdk.Geo.Core.Models;

public class NotificareVisit
{
    public DateTime DepartureDate { get; }
    public DateTime ArrivalDate { get; }
    public double Latitude { get; }
    public double Longitude { get; }

    public NotificareVisit(DateTime departureDate, DateTime arrivalDate, double latitude, double longitude)
    {
        DepartureDate = departureDate;
        ArrivalDate = arrivalDate;
        Latitude = latitude;
        Longitude = longitude;
    }
}

File: Notificare.Geo.iOS.Binding.Bridge/Sources/Models/NotificareVisit.swift
import Foundation
import NotificareGeoKit

@objc
public class NotificareVisit : NSObject {
    @objc public let departureDate: Date
    @objc public let arrivalDate: Date
    @objc public let latitude: Double
    @objc public let longitude: Double

    @objc public init(departureDate: Date, arrivalDate: Date, latitude: Double, longitude: Double) {
        self.departureDate = departureDate
        self.arrivalDate = arrivalDate
        self.latitude = latitude
        self.longitude = longitude
    }

    public convenience init(from visit: NotificareGeoKit.NotificareVisit) {
        self.init(
            departureDate: visit.departureDate,
            arrivalDate: visit.arrivalDate,
            latitude: visit.latitude,
            longitude: visit.longitude
        )
    }
}
